codegenerator/cli/templates/static/codegen/src/db/DbFunctions.res

```diff
@@ -287,101 +259,239 @@ module EntityHistory = {
+
+    This is needed since unordered multichain mode only cares about any changes that 
+    occurred after the first change on the reorg chain. To prevent skipping or double processing events
+    on the other chains. If for instance there are no entity changes based on the reorg chain, the other
+    chains do not need to be rolled back, and if the reorg chain has new included events, it does not matter
+    that if those events are processed out of order from other chains since this is "unordered_multichain_mode"
+
+    Ordered multichain mode needs to ensure that all chains rollback to any event that occurred after the reorg chain
+    block number. Regardless of whether the reorg chain incurred any changes or not to entities.
+    */
+    let makeGetFirstChangeSerial = (self: t, ~entityName) =>
+      switch self {
+      | OrderedMultichain({safeBlockTimestamp, reorgChainId, safeBlockNumber}) =>
+        sql =>
+          OrderedMultichain.getFirstChangeSerial(
+            sql,
+            ~safeBlockTimestamp,
+            ~reorgChainId,
+            ~safeBlockNumber,
+            ~entityName,
+          )
+      | UnorderedMultichain({reorgChainId, safeBlockNumber}) =>
+        sql =>
+          UnorderedMultichain.getFirstChangeSerial(
+            sql,
+            ~reorgChainId,
+            ~safeBlockNumber,
+            ~entityName,
+          )
+      }
+
+    let getLogger = (self: t, ~entityName) => {
+      switch self {
+      | OrderedMultichain({safeBlockTimestamp, reorgChainId, safeBlockNumber}) =>
+        Logging.createChild(
+          ~params={
+            "type": "OrderedMultichain",
+            "safeBlockTimestamp": safeBlockTimestamp,
+            "reorgChainId": reorgChainId,
+            "safeBlockNumber": safeBlockNumber,
+            "entityName": entityName,
+          },
+        )
+      | UnorderedMultichain({reorgChainId, safeBlockNumber}) =>
+        Logging.createChild(
+          ~params={
+            "type": "UnorderedMultichain",
+            "reorgChainId": reorgChainId,
+            "safeBlockNumber": safeBlockNumber,
+            "entityName": entityName,
+          },
+        )
+      }
+    }
   }
 
-  let rollbackDiffResponse_decode = (json: Js.Json.t) => {
-    json
-    ->S.parseWith(rollbackDiffResponseRawSchema)
-    ->Belt.Result.flatMap(raw => {
-      switch raw {
-      | {
-          val: Some(val),
-          chain_id: Some(chainId),
-          block_number: Some(blockNumber),
-          block_timestamp: Some(blockTimestamp),
-          log_index: Some(logIndex),
-          entity_type,
-        } =>
-        Entities.getEntityParamsDecoder(entity_type)(val)->Belt.Result.map(entity => {
-          let eventIdentifier: Types.eventIdentifier = {
-            chainId,
-            blockTimestamp,
-            blockNumber,
-            logIndex,
-          }
-
-          Some({entity, eventIdentifier})
+  let deleteAllEntityHistoryAfterEventIdentifier = (~isUnorderedMultichainMode) => (
```

Comment:
This just kept it's naming/signature but I plan on doing some refactoring for all the delete functions that happen on a rollback in the next PR